docs: Plugin docs cleanup & expansion (#16862)
eslint/eslint: docs/src/extend/plugins.md

### Configs in Plugins

You can bundle configurations inside a plugin by specifying them under the `configs` key. This can be useful when you want to bundle a set of custom rules with additional configuration. Multiple configurations are supported per plugin.

> It is not possible to specify a default configuration for a given plugin. Users must specify a default configuration in their configuration file.

Reviewer: "Users _must_ specify a _default_ configuration in their configuration file." sounds confusing.

* "must" - sounds like users are required to use a configuration from the plugin, but they're not.
* "default" - if the user wants to use a configuration from the plugin, they specify it in their configuration file so there's no default.

            }
        }
    }
    rules: {
        "my-rule": {/* rule definition */},
        "another-rule": {/* rule definition */},
        "yet-another-rule": {/* rule definition */},
    }
};
```

> Plugins cannot force a specific configuration to be used. Users must manually include plugin's configurations in their configuration file.

Reviewer 2: ```suggestion
Plugins cannot force a specific configuration to be used. Users must manually include the plugin's configurations in their configuration file.
```

Author: ```suggestion
Plugins cannot force a specific configuration to be used. Users must manually include a plugin's configurations in their configuration file.
```

If the example plugin above were called `eslint-plugin-myPlugin`, the `myConfig` and `myOtherConfig` configurations would then be usable in a configuration file by extending `"plugin:myPlugin/myConfig"` and `"plugin:myPlugin/myOtherConfig"`, respectively.

```json
// eslintrc.json

{
    "extends": ["plugin:myPlugin/myConfig"]
}

```

### Peer Dependency

> To make clear that the plugin requires ESLint to work correctly, you must declare ESLint as a `peerDependency` in your plugin's `package.json`.
> 
> Plugin support was introduced in ESLint version `0.8.0`. Ensure the `peerDependency` points to ESLint `0.8.0` or later.

Reviewer 2: The official name of the key is package.json is [`peerDependencies` ](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependencies). We should use this to avoid any confusion. What do you think? 

```suggestion
To make clear that the plugin requires ESLint to work correctly, you must declare ESLint as a peer dependency by mentioning it in the `peerDependencies` field of your plugin's `package.json`.

Plugin support was introduced in ESLint version `0.8.0`. Ensure the `peerDependencies` points to ESLint `0.8.0` or later.
```

```

> ESLint configuration files do not enable any of the plugin's rules by default. You must specify your plugin name in the `plugins` array as well as any rules you want to enable that are part of the plugin. Any plugin rules must be prefixed with the short or long plugin name. To learn more, refer to [Configure Plugins](../use/configure/plugins#configure-plugins).

Reviewer: The first sentence was referring to configs in plugins, not end user config files.

Author: i'm not 100% sure if i understood what you meant there, but i think i refactored per your feedback in [bf166a3](https://github.com/eslint/eslint/pull/16862/commits/bf166a3a39148d0d6b3ae579ba9165fdf6e6a1a9). please let me know if you'd like additional changes.

> If you'd like to include individual rules from a plugin in your configuration file, you must specify your plugin name in the `plugins` array as well as any rules you want to enable that are part of the plugin. Any plugin rules must be prefixed with the short or long plugin name. To learn more, refer to [Configure Plugins](../use/configure/plugins#configure-plugins).

Author: ahh ok this is much clearer to me now. i will rephrase.

> If you'd like to include individual rules from a plugin in your configuration file, you must specify your plugin name in the `plugins` array as well as any rules you want to enable that are part of the plugin. Any plugin rules must be prefixed with the short or long plugin name. To learn more, refer to [Configure Plugins](../use/configure/plugins#configure-plugins).

Author: ok, rephrased and reorded this whole section based on this feedback. changes added in [e5bf603](https://github.com/eslint/eslint/pull/16862/commits/e5bf60368652a5726e7533905b21bc83f12f7924).

i think the whole section is much clearer now.

### Peer Dependency

> To make clear that the plugin requires ESLint to work correctl, you must declare ESLint as a `peerDependency` in your plugin's `package.json`.

Reviewer 3: ```suggestion
To make clear that the plugin requires ESLint to work correctly, you must declare ESLint as a `peerDependency` in your plugin's `package.json`.
```
